### DISCUSSION TITLE
Session is null after logging in

### DISCUSSION THREAD

**Discussion by paulmbw** [supabase/supabase#12524] [Questions] [answered]
# Bug report
Related to [this](https://github.com/supabase/supabase/issues/6764) which was closed.

## Describe the bug
After logging in and retrieving the session using `supabase.auth.getSession()`, the session is always null. I'm using `2.8.0` of `@supabase/supabase-js`

This is what my `_app.tsx` looks like:

```
export default function App({ Component, pageProps }: AppProps) {
  const [supabase] = useState(() => createBrowserSupabaseClient());

  return (
    <SessionContextProvider
      supabaseClient={supabase}
      initialSession={pageProps.initialSession}
    >
      <Component {...pageProps} />
    </SessionContextProvider>
  );
}
```

This is my component for getting the session of the logged in user:
```
import { createServerSupabaseClient } from "@supabase/auth-helpers-nextjs";

const Home = ({ user }: any) => {
  return <div><h1>Welcome, {user.email}}</h1></div>
};

export const getServerSideProps = async (ctx: any) => {
  // Create authenticated Supabase Client
  const supabase = createServerSupabaseClient(ctx);
  // Check if we have a session
  const {
    data: { session },
  } = await supabase.auth.getSession();

  console.log({ session });

  if (!session) {
    return {
      redirect: {
        destination: "/signin",
        permanent: false,
      },
    };
  }

  return {
    props: {
      initialSession: session,
      user: session.user,
    },
  };
};

export default Home;
```

And this is the logic when I am logging in:

```
  const { data } = await supabase.auth.signInWithPassword({
        email,
        password,
      });

      if (data) {
        return router.push("/");
      }
```

What am I missing?

**Comment by paulmbw** [comment]
Forgot to mention, this is how I am creating the supabase client:

```
import { createClient } from "@supabase/supabase-js";

const supabaseUrl = process.env.NEXT_PUBLIC_SUPABASE_URL || "";
const supabaseAnonKey = process.env.NEXT_PUBLIC_SUPABASE_ANON_KEY || "";

const supabase = createClient(supabaseUrl, supabaseAnonKey);

export default supabase;
```

@GaryAustin1 @kiwicopple Any idea what's going wrong here?

**Comment by GaryAustin1** [comment]
It looks like you are using auth-helpers and although it can be finicky, I don't think the sign in process has this sort of bug if used correctly.   There is an issue if you use createClient with supabase-js on a server directly without doing persistSession:false, but auth-helpers does that.

I really recommend you use Supabase Discord to get some help with this, as it is likely setup/use issue and not a bug without more work to isolate.  I don't use auth-helpers, but users in the Discord do.

Thanks

**Comment by paulklen** [comment]
> It looks like you are using auth-helpers and although it can be finicky, I don't think the sign in process has this sort of bug if used correctly.   There is an issue if you use createClient with supabase-js on a server directly without doing persistSession:false, but auth-helpers does that.
> 
> 
> 
> I really recommend you use Supabase Discord to get some help with this, as it is likely setup/use issue and not a bug without more work to isolate.  I don't use auth-helpers, but users in the Discord do.
> 
> 
> 
> Thanks

Thanks a lot Gary! I've posted a message on discord here 👉🏾 https://discord.com/channels/839993398554656828/1076557461865115759

Hopefully will get a reply soon

**Comment by GaryAustin1** [comment]
Moved this to discussion as it is likely an issue with using a supabase-js client on the browser instead of the auth-helpers browser client.
https://supabase.com/docs/guides/auth/auth-helpers/nextjs-server-components#client-side
It is also now on Discord so can be followed up either place.
If it turns out to be a bug will move back.

**Comment by paulmbw** [comment, marked as answer]
Thanks Gary, `@supabase/auth-helpers-nextjs` in fact solved this. This is how I am creating the supabase client now:

```
import { createBrowserSupabaseClient } from "@supabase/auth-helpers-nextjs";

const supabaseClient = createBrowserSupabaseClient();

export default supabaseClient;
```

Free to close this discussion. Thanks for your help, and enjoy the rest of your weekend!
